### ISSUE TITLE
parse when building？

### ISSUE THREAD

**Issue by AaronMarsLuo** [hankcs/AhoCorasickDoubleArrayTrie#44]
Is it safe to parse word when ACDAT is building?
we got a wrong parsing result when the value word is not in the origin text.
we doubt whether it's safe to parse when building.

**Comment by hankcs** [comment]
No, not safe at all. How could it be safe when its states are getting updated during building?
